packages/cli/src/databases/migrations/common/1717498465932-RefactorExecutionIndices.ts

```diff
@@ -0,0 +1,87 @@
+import type { MigrationContext, ReversibleMigration } from '@/databases/types';
+
+/**
+ * Add new indices:
+ *
+ * - `status, startedAt` for default query at `ExecutionRepository.findManyByRangeQuery`
+ * - `workflowId, status, startedAt` for filtered query at `ExecutionRepository.findManyByRangeQuery`
+ * - `waitTill, status` for regular queries at `ExecutionRepository.getWaitingExecutions`
+ *
+ * Remove unused indices in sqlite:
+ *
+ * - `stoppedAt` (duplicate with different casing)
+ * - `waitTill`
+ * - `status, workflowId`
+ *
+ * Remove unused indices in MySql:
+ *
+ * - `status`
+ *
+ * Remove unused indices in all DBs:
+ *
+ * - `stopped_at`
+ * - `waitTill, id`
+ * - `workflowId, id`
+ *
+ * Keep index as is:
+ *
+ * - `deletedAt` for query at `ExecutionRepository.hardDeleteSoftDeletedExecutions`
+ */
+export class RefactorExecutionIndices1717498465932 implements ReversibleMigration {
+	async up({ schemaBuilder, isPostgres, isSqlite, isMysql }: MigrationContext) {
+		await schemaBuilder.createIndex('execution_entity', ['status', 'startedAt']);
+		await schemaBuilder.createIndex('execution_entity', ['workflowId', 'status', 'startedAt']);
+		await schemaBuilder.createIndex('execution_entity', ['waitTill', 'status']);
+
+		if (isSqlite) {
+			await schemaBuilder.dropIndex('execution_entity', ['stoppedAt'], {
```

Comment:
could we combine this with the other `dropIndex` on `stoppedAt` ?

Reply:
[059ddf9](https://github.com/n8n-io/n8n/pull/9817/commits/059ddf984c067a050e03f6bd2ab008319208df5e)